gpu/internal/opengl/opengl.go

```diff
@@ -359,10 +359,12 @@ func (b *Backend) queryState() glState {
 func (b *Backend) restoreState(dst glState) {
 	src := b.glstate
 	f := b.funcs
+
+	prev := src.texUnits.active
 	for i, unit := range dst.texUnits.binds {
 		src.bindTexture(f, i, unit)
 	}
-	src.activeTexture(f, dst.texUnits.active)
+	src.activeTexture(f, prev)
```

Comment:
When activeTexture is called, it will make any subsequent calls to bindTexture to assign a texture of specific binding type to that texture unit.
The function to obtain current gl state (queryState), is working correctly:
+ grabs currently active texture and saves it to gltextUnits.active.
+ grabs currently bounds texture to unit textures 0 and 1 and saves them to texUnits.binds

The problem comes when resetting the gl state to its original state (restoreState).
+ eventually resetting texture units in texUnits.binds, calls activeTexture. Since there are 2 bindings, the end call is activeTexture(TEXTURE1). This makes subsequent bindTexture calls to bind to texture unit 1, and not the value obtained in saveState. On the game side, the uniform sampler expects to be bount to TEXTURE0. so GIO is not cleaning after itself properly.

The fix, resets activeTexture to whatever value was saved originally here: https://github.com/gioui/gio/blob/0867ab229f640d42e189c0dff078d6a8ecf77b91/gpu/internal/opengl/opengl.go#L324. 
Since activeTexture function relies on the value of glTexUnhits.active, we need to save textUnits.active value, before resetting the two textures, or otherwise gets overwritten, and left in the wrong state.

This fix simply puts the gl state correctly, as it was before calling queryState.